tests/test_scheduler_worker.py

```diff
@@ -57,19 +57,30 @@ def test_mat_add(scheduler_client, matrix1, matrix2):
 
 def test_mat_subtract(scheduler_client, matrix1, matrix2):
     task = Task(taskId="2", taskDefintion="mat_subtract", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()])
-    future = scheduler_client.SubmitTask(task)
+    future = scheduler_client.SubmitTask(task)[0]
 
     worker_client = WorkerClient(future.hostName, future.port)
 
     expected_result = [[-4, -4 ], [-4, -4]]
     poll_for_result(worker_client, future, expected_result)
 
 def test_passing_futures_as_args_flow(scheduler_client, matrix1, matrix2):
-    future = scheduler_client.SubmitTask(Task(taskId="0", taskDefintion="dot_product", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))
-    future2 = scheduler_client.SubmitTask(Task(taskId="1", taskDefintion="mat_add", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))
-    future3 = scheduler_client.SubmitTask(Task(taskId="2", taskDefintion="mat_subtract", taskData=[future, future2]))
+    future = scheduler_client.SubmitTask(Task(taskId="0", taskDefintion="dot_product", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))[0]
+    future2 = scheduler_client.SubmitTask(Task(taskId="1", taskDefintion="mat_add", taskData=[json.dumps(matrix1).encode(), json.dumps(matrix2).encode()]))[0]
+    future3 = scheduler_client.SubmitTask(Task(taskId="2", taskDefintion="mat_subtract", taskData=[future, future2]))[0]
 
     workerClient = WorkerClient(future3.hostName, future3.port)
 
     expected_result = [[13, 14], [33, 38]]
     poll_for_result(workerClient, future3, expected_result)
+
+# Multiple workers here doensn't guarantee that the task will be assigned to different workers. It depends on the scheduling mode and the number of workers registered.
```

Comment:
if number of workers registered =AssingedWorkersPerTask. Then does it guarantee?